### PR TITLE
feat: DialogLayout

### PR DESCRIPTION
**Pull request by lukelooker**
### :sparkles: Changes

Standard composition of `Dialog` is:

- `DialogHeader`, `DialogContent`, `DialogFooter`

This PR addresses support for less common `Dialog` compositions:

- Just `DialogContent`
- `DialogHeader`, `DialogContent`
- `DialogContent`, `DialogFooter`

By introducing a new `DialogLayout` "helper" component that manages adding new `hasHeader` and `hasFooter` props to the `DialogContent` component.

Also updates existing usages of `DialogContent` to use `DialogLayout` helper consistently.

Updates screenshots for `DataTable` that changed extremely subtly for Safari fix (change is not user perceptible)

### :white_check_mark: Requirements

- [ ] Includes test coverage for all changes
- [ ] Build and tests are passing
- [ ] Update documentation
- [ ] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [ ] Check for image-snapshot changes (run `yarn image-snapshots` locally)
- [ ] PR is ideally < ~400LOC

### :camera: Screenshots

Header-snap
Footer-snap
Basic-snap

### CODE REVIEW

File: packages/components/src/Dialog/Layout/DialogLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  header,
+  headerCloseButton = !footer && true,
+  headerDetail,
+  isLoading,
+}) => {
+  const dialogHeader = header && (
+    <DialogHeader hideClose={!headerCloseButton} detail={headerDetail}>
+      {header}
+    </DialogHeader>
+  )
```
Comment (lukelooker) [inline packages/components/src/Dialog/Layout/DialogLayout.tsx:82]: If the `Dialog` has a footer we assume the footer includes affordances to close the `Dialog` and remove the "x" in the header (future-proofing for a world where that's no longer a supported pattern)

File: packages/components/src/Dialog/Layout/DialogLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+  const dialogFooter = (footer || footerSecondary) && (
+    <DialogFooter secondary={footerSecondary}>{footer}</DialogFooter>
+  )
+
+  return (
+    <>
+      {dialogHeader}
+      <DialogContent hasFooter={!dialogFooter} hasHeader={!dialogHeader}>
+        {isLoading ? <DialogLoading /> : children}
```
Comment (lukelooker) [inline packages/components/src/Dialog/Layout/DialogLayout.tsx:92]: Built-in loading pattern :)

**Review by mdodgelooker — COMMENTED** [review]
I think this might need a rebase from the latest `main`. The sandboxes in gatsby aren't working right:
image

File: packages/components/src/Dialog/Layout/DialogLayout.tsx (hunk 2 above)
Comment (mdodgelooker) [reply packages/components/src/Dialog/Layout/DialogLayout.tsx:92]: Right on!

File: packages/components/src/Dialog/stories/SaveChanges.tsx [hunk 3]
```diff
@@ -87,31 +81,29 @@ export const SaveChanges = () => {
...
+          <DialogLayout footer={footer} header="Edit Name">
+            <Form onSubmit={handleSave}>
+              <FieldText
+                label="Name"
+                onChange={handleChange}
+                value={name}
+                description='Enter name, then click "Cancel" to trigger confirmation'
+              />
+            </Form>
+          </DialogLayout>
```
Comment (mdodgelooker) [inline packages/components/src/Dialog/stories/SaveChanges.tsx:106]: It's nice that you don't need the React Fragment anymore in this kind of composition. 🙂

File: www/src/documentation/components/dialogs/layout.mdx [hunk 4]
```diff
@@ -4,6 +4,16 @@ storybook: true
 storybookPath: DialogContent
 ---
 
+## DialogLayout
+
+`DialogLayout`
+
+```jsx
+<DialogLayout header="Dialog Header" footer="DialogFooter">
+  This is a DialogHeader
```
Comment (mdodgelooker) [inline www/src/documentation/components/dialogs/layout.mdx:13]:
```suggestion
  This is a DialogLayout
```
Should this section include content on when to use `DialogLayout` vs `DialogContent`?

**Review by mdodgelooker — CHANGES_REQUESTED** [review]
Looks great! Just `layout.mdx` needs a lil love.

File: www/src/documentation/components/dialogs/layout.mdx (hunk 4 above)
Comment (lukelooker) [reply www/src/documentation/components/dialogs/layout.mdx]: Yep! Fixing.

**Comment by lukelooker** [comment]
@mdodgelooker Broke apart the `Dialog Layout` page into per-component pages and improved the content on them :)

**Review by mdodgelooker — COMMENTED** [review]
LGTM!

**Review by mdodgelooker — APPROVED** [review]

**Review by mdodgelooker — APPROVED** [review]

**Pull request merged**